src/pages/Auth/Signup.jsx

```diff
@@ -1,7 +1,189 @@
-import React from "react";
+import React, { useCallback, useEffect, useState } from "react";
+import { constants, debounce, handleChange } from "../../helpers";
+import { Link, useNavigate } from "react-router-dom";
+import Input from "./components/Input";
+import "./Auth.css";
+import { signupUser, userNameExists } from "./helpers";
+import { getDownloadURL, ref, uploadBytesResumable } from "firebase/storage";
+import { storage } from "../../firebase-config";
+import { Brand } from "../../components";
+import { FiUpload } from "react-icons/fi";
 
 const Signup = () => {
-  return <div>Signup</div>;
+  const navigate = useNavigate();
+  const [isValidDetails, setisValidDetails] = useState(true);
+  const [isValidUsername, setIsValidUsername] = useState(true);
```

Comment:
Have a look into formik + yup for form validations. Would make life easier.

Reply:
Yes will migrate to formik once authentication is completely done